lib/mzgl/gl/Graphics.cpp

```diff
@@ -572,140 +365,34 @@ void Graphics::drawArc(glm::vec2 c, float r, float startAngle, float endAngle) {
-	glVertexAttribPointer(currShader->positionAttribute, 2, GL_FLOAT, GL_FALSE, 0, NULL);
-
-	glEnableVertexAttribArray(currShader->colorAttribute);
-	glBindBuffer(GL_ARRAY_BUFFER, immediateColorBuffer);
-	glBufferData(GL_ARRAY_BUFFER, cols.size() * 4 * sizeof(float), cols.data(), GL_DYNAMIC_DRAW);
-	glVertexAttribPointer(currShader->colorAttribute, 4, GL_FLOAT, GL_FALSE, 0, NULL);
-
-	glDrawArrays(primitiveTypeToGLMode(type), 0, (GLsizei) verts.size());
-
-	glDisableVertexAttribArray(currShader->positionAttribute);
-	glDisableVertexAttribArray(currShader->colorAttribute);
-
-#ifndef MZGL_GL2
-	glBindVertexArray(0);
-#endif
-}
-
 void Graphics::setBlendMode(BlendMode blendMode) {
-	if (blendMode == BlendMode::Additive) {
-		glBlendFunc(GL_SRC_ALPHA, GL_ONE);
-	} else {
-		glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
-	}
+	api->setBlendMode(blendMode);
 }
 
-void Graphics::drawVerts(const vector<glm::vec2> &verts, const vector<uint32_t> &indices) {
-	//	VboRef vbo = Vbo::create();
-	//	vbo->setVertices(verts);
-	//	vbo->setIndices(indices);
-	//	vbo->draw(*this);
-
-	nothingShader->begin();
-	currShader->uniform("mvp", getMVP());
-	if (currShader->needsColorUniform) {
-		currShader->uniform("color", getColor());
-	}
-	// now draw
-
-	Log::e() << "drawVerts(verts, indices) has changed and is not tested";
-
-#ifndef MZGL_GL2
-	if (immediateVertexArray == 0) return;
-	glBindVertexArray(immediateVertexArray);
-#endif
-
-	glEnableVertexAttribArray(currShader->positionAttribute);
-	glBindBuffer(GL_ARRAY_BUFFER, immediateVertexBuffer);
-	glBufferData(GL_ARRAY_BUFFER, verts.size() * 2 * sizeof(float), verts.data(), GL_DYNAMIC_DRAW);
-	glVertexAttribPointer(currShader->positionAttribute, 2, GL_FLOAT, GL_FALSE, 0, NULL);
-
-	//	glEnableVertexAttribArray(currShader->);
-	glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, immediateColorBuffer);
-	glBufferData(GL_ELEMENT_ARRAY_BUFFER, indices.size() * sizeof(uint32_t), indices.data(), GL_DYNAMIC_DRAW);
-	//	glVertexAttribPointer(currShader->colorAttribute, 4, GL_FLOAT, GL_FALSE, 0, NULL);
-
-	glDrawElements(GL_TRIANGLES, (GLsizei) indices.size(), GL_UNSIGNED_INT, 0);
-	//	glDrawArrays(type, 0, verts.size());
-
-	glDisableVertexAttribArray(currShader->positionAttribute);
-	//	glDisableVertexAttribArray(currShader->colorAttribute);
+void Graphics::drawVerts(const std::vector<glm::vec2> &verts,
+						 const std::vector<glm::vec4> &cols,
+						 Vbo::PrimitiveType type) {
+	api->drawVerts(verts, cols, type);
+}
 
-#ifndef MZGL_GL2
-	glBindVertexArray(0);
-#endif
+void Graphics::drawVerts(const std::vector<glm::vec2> &verts, Vbo::PrimitiveType type) {
+	api->drawVerts(verts, type);
 }
 
 void Graphics::maskOn(const Rectf &r) {
-	glEnable(GL_SCISSOR_TEST);
-	glScissor(r.x, height - (r.bottom()), r.width, r.height);
+	api->maskOn(r);
 }
 
 void Graphics::maskOff() {
-	glDisable(GL_SCISSOR_TEST);
+	api->maskOff();
 }
 
 bool Graphics::isMaskOn() {
-	return glIsEnabled(GL_SCISSOR_TEST);
+	return api->isMaskOn();
```

Comment:
This just made me think of :

![gif](https://github.com/elf-audio/mzgl/assets/12952415/9b04ab3a-f28d-46aa-b346-ac9f22af7d0a)

LOL